In [1]:
import pandas as pd

In [2]:
df = pd.read_excel('Unlinked Metadata 0625.xlsx')
df.head()

Subject ID Patient Consented Sample Type_#   CGA  DOL   Current Weight  \
0    NB00012                 N   NB00012_M_1  28.4    27            1430   
1    NB00003                 Y   NB00003_M_1  29.0    22             995   
2    NB00003                 Y   NB00003_M_2  29.0    22            1005   
3    NB00003                 Y   NB00003_M_3  29.3    25            1065   
4    NB00354                 N   NB00354_M_1  29.4    12            1005   

   Current Height  Current HC Scavenged/Fresh?   MBM/DMB? HMF Y/N? TPN Y/N?  \
0            32.0        26.0       Scavenged   MBM + DBM        Y      NaN   
1            35.7        23.7       Scavenged         DBM        Y        N   
2            36.0        24.4       Scavenged         DBM        Y        N   
3            36.0        24.3       Scavenged         DBM        Y        N   
4            34.0        34.5       Scavenged         MBM        Y        Y   

  Iron Y/N?  Linked?  # Aliquots                      Additional Comments  
0        NaN        N    5 (~0.8)  Residual from Milk Prep Room & B2 Full  
1          Y        N           2                  Scavenged Feeding Tube  
2          Y        N           1                  Scavenged Feeding Tube  
3          Y        N           1                  Scavenged Feeding Tube  
4        NaN        N           7            Residual from Milk Prep Room

In [3]:
AC = pd.read_excel('NeoBank Unlinked AC.xlsx')
AC.head()

sample ID  inj vol          2'FL  Iso (control)         DFLAC  \
0  NB00003_M_27       15           NaN            NaN           NaN   
1  NB00003_M_26       15  1.051478e+06   7.053702e+06  1.801466e+05   
2    NB0010_M_4       15  2.224422e+07   6.946214e+06  1.466091e+06   
3   NB00004_M_4       15  3.213107e+07   7.087612e+06  1.108820e+05   
4   NB00014_M_1       15  4.736746e+07   7.159182e+06  7.137924e+06   

           3'SL          6'SL           LNT          LNnT         LNFPI  \
0           NaN           NaN           NaN           NaN           NaN   
1  2.556277e+05  3.403535e+05  2.309651e+05  1.091730e+05  2.635778e+05   
2  1.403590e+06  5.509796e+06  8.950674e+06  6.919087e+05  6.229519e+06   
3  7.738709e+05  3.660383e+06  6.932930e+06  1.667270e+06  1.513378e+07   
4  2.049916e+06  5.323517e+06  7.671713e+06  2.415186e+06  1.587956e+07   

         LNFPII      LNFPIII          LSTc         DFLNT         DSLNT  \
0           NaN          NaN           NaN           NaN           NaN   
1  2.057756e+05  101484.2646  2.069264e+05  1.724065e+05  1.696447e+05   
2  5.821316e+06  182680.4001  1.378025e+06  5.670600e+06  2.037842e+06   
3  1.778976e+05  199407.6536  1.938777e+06  4.739951e+05  1.129592e+06   
4  2.630392e+06  365193.0543  4.721838e+06  5.694380e+06  1.872350e+06   

         DFLNH       FDSLNH         DSLNH  
0          NaN          NaN           NaN  
1   99897.0777  172129.7542  1.006923e+05  
2  449381.8137  810615.6873  1.103405e+06  
3  409684.5743  233999.6664  4.959159e+05  
4  274316.2401  345379.6165  1.061841e+06

In [4]:
df = df.rename(columns={"Sample Type_#": "sample_unique_id"})
AC = AC.rename(columns={"sample ID": "sample_unique_id"})


In [5]:
df['sample_unique_id'] = df['sample_unique_id'].str.strip()
AC['sample_unique_id'] = AC['sample_unique_id'].str.strip()

In [6]:
AC

sample_unique_id  inj vol          2'FL  Iso (control)         DFLAC  \
0       NB00003_M_27       15           NaN            NaN           NaN   
1       NB00003_M_26       15  1.051478e+06   7.053702e+06  1.801466e+05   
2         NB0010_M_4       15  2.224422e+07   6.946214e+06  1.466091e+06   
3        NB00004_M_4       15  3.213107e+07   7.087612e+06  1.108820e+05   
4        NB00014_M_1       15  4.736746e+07   7.159182e+06  7.137924e+06   
..               ...      ...           ...            ...           ...   
130      NB00380_M_5       15  2.548159e+06   8.374666e+06  9.963675e+04   
131      NB00438_M_3       15  3.714314e+07   8.849808e+06  3.128912e+06   
132      NB00457_M_1       15  7.070979e+07   9.802165e+06  5.448618e+06   
133      NB00468_M_1       15  3.233975e+07   9.937304e+06  3.918450e+06   
134      NB00457_M_2       15  4.627295e+07   1.087569e+07  3.928213e+06   

             3'SL          6'SL           LNT          LNnT         LNFPI  \
0             NaN           NaN           NaN           NaN           NaN   
1    2.556277e+05  3.403535e+05  2.309651e+05  1.091730e+05  2.635778e+05   
2    1.403590e+06  5.509796e+06  8.950674e+06  6.919087e+05  6.229519e+06   
3    7.738709e+05  3.660383e+06  6.932930e+06  1.667270e+06  1.513378e+07   
4    2.049916e+06  5.323517e+06  7.671713e+06  2.415186e+06  1.587956e+07   
..            ...           ...           ...           ...           ...   
130  2.203344e+06  6.868806e+06  1.093594e+07  2.809632e+06  1.106507e+06   
131  1.859451e+06  4.517493e+06  7.412311e+06  1.563939e+06  1.331367e+07   
132  3.358550e+06  1.142248e+07  2.260011e+07  2.418345e+06  3.402642e+07   
133  2.446762e+06  3.289483e+06  4.012047e+06  9.774770e+05  4.947740e+06   
134  1.984679e+06  1.820996e+07  1.579394e+07  1.933105e+06  2.520000e+07   

           LNFPII       LNFPIII          LSTc         DFLNT         DSLNT  \
0             NaN           NaN           NaN           NaN           NaN   
1    2.057756e+05  1.014843e+05  2.069264e+05  1.724065e+05  1.696447e+05   
2    5.821316e+06  1.826804e+05  1.378025e+06  5.670600e+06  2.037842e+06   
3    1.778976e+05  1.994077e+05  1.938777e+06  4.739951e+05  1.129592e+06   
4    2.630392e+06  3.651931e+05  4.721838e+06  5.694380e+06  1.872350e+06   
..            ...           ...           ...           ...           ...   
130  1.156049e+07  1.443196e+05  3.168754e+06  1.494014e+06  1.248896e+06   
131  3.463945e+06  1.708952e+05  2.663312e+06  5.194955e+06  9.190932e+05   
132  7.158418e+06  1.070328e+06  8.963838e+06  1.564165e+07  3.054897e+06   
133  5.697583e+06  1.941846e+05  1.292248e+06  5.087703e+06  9.968905e+05   
134  6.228110e+06  6.749985e+05  6.847580e+06  1.200914e+07  2.249618e+06   

           DFLNH        FDSLNH         DSLNH  
0            NaN           NaN           NaN  
1     99897.0777  1.721298e+05  1.006923e+05  
2    449381.8137  8.106157e+05  1.103405e+06  
3    409684.5743  2.339997e+05  4.959159e+05  
4    274316.2401  3.453796e+05  1.061841e+06  
..           ...           ...           ...  
130  837453.6015  2.066399e+06  2.087714e+06  
131  387868.2138  3.366231e+05  1.031563e+06  
132  474998.2730  2.857115e+05  1.594684e+06  
133  662102.2885  6.792672e+05  6.872472e+05  
134  456776.0545  4.449340e+05  1.598231e+06  

[135 rows x 18 columns]

In [7]:
merged = pd.merge(df, AC, on="sample_unique_id", how="inner")


In [8]:
merged

Subject ID Patient Consented sample_unique_id   CGA  DOL   Current Weight  \
0      NB00012                 N      NB00012_M_1  28.4    27            1430   
1      NB00003                 Y      NB00003_M_1  29.0    22             995   
2      NB00003                 Y      NB00003_M_2  29.0    22            1005   
3      NB00003                 Y      NB00003_M_3  29.3    25            1065   
4      NB00354                 N      NB00354_M_1  29.4    12            1005   
..         ...               ...              ...   ...   ...             ...   
117    NB00007                 Y      NB00007_M_2  38.4    12            2190   
118    NB00007                 Y      NB00007_M_3  39.2    17            2340   
119    NB00007                 Y      NB00007_M_4  39.3    18            2380   
120    NB00379                 Y      NB00379_M_1  39.6    98            3073   
121    NB00007                 Y      NB00007_M_5  40.0    23            2430   

     Current Height  Current HC Scavenged/Fresh?   MBM/DMB?  ...  \
0              32.0        26.0       Scavenged   MBM + DBM  ...   
1              35.7        23.7       Scavenged         DBM  ...   
2              36.0        24.4       Scavenged         DBM  ...   
3              36.0        24.3       Scavenged         DBM  ...   
4              34.0        34.5       Scavenged         MBM  ...   
..              ...         ...              ...        ...  ...   
117            44.0        32.5       Scavenged         MBM  ...   
118            44.0        33.0       Scavenged         MBM  ...   
119            44.0        33.0       Scavenged         MBM  ...   
120            45.0        33.2       Scavenged         MBM  ...   
121            45.0        34.5       Scavenged        MBM   ...   

             LNnT         LNFPI        LNFPII      LNFPIII          LSTc  \
0    7.026127e+05  3.686662e+06  9.449716e+06  115974.6422  1.878166e+06   
1    3.554526e+05  2.188585e+06  2.289321e+06   75037.6026  4.920676e+05   
2    1.009692e+06  4.518359e+06  3.409862e+06   82939.5756  7.308408e+05   
3    6.291601e+05  4.624398e+06  3.044566e+06   95822.1100  6.416626e+05   
4    3.386165e+05  7.478814e+05  7.146520e+06  167202.2557  1.778718e+06   
..            ...           ...           ...          ...           ...   
117  1.792379e+06  9.377929e+06  6.286603e+06  347147.3171  3.210144e+06   
118  6.844752e+05  6.025784e+06  4.126527e+06  174538.3580  1.167148e+06   
119  5.297594e+05  2.120913e+06  1.480291e+07  532044.6917  3.070643e+06   
120  6.555827e+05  1.314651e+07  1.863176e+06   84150.5867  1.410230e+06   
121  8.896175e+05  6.627449e+06  7.549175e+06  211982.9758  1.452051e+06   

            DFLNT         DSLNT         DFLNH        FDSLNH         DSLNH  
0    2.210782e+06  1.358187e+06  1.686614e+06  2.734042e+06  3.015652e+06  
1    2.433375e+06  4.284901e+05  2.886016e+05  2.508040e+05  2.211292e+05  
2    2.628986e+06  5.605861e+05  6.504227e+05  5.128080e+05  5.097773e+05  
3    1.450763e+06  4.879326e+05  4.226243e+05  7.237869e+05  4.211949e+05  
4    9.524693e+05  1.128755e+06  1.396158e+06  2.356293e+06  2.505246e+06  
..            ...           ...           ...           ...           ...  
117  6.198172e+06  1.764609e+06  5.750102e+05  8.405371e+05  1.400118e+06  
118  5.337920e+06  1.809488e+06  1.026485e+06  2.662468e+05  8.587586e+05  
119  1.352836e+06  1.923140e+06  5.688529e+05  1.727998e+06  1.916662e+06  
120  5.056181e+04  5.865819e+05  5.460105e+05  1.194098e+05  1.291120e+06  
121  6.723469e+06  2.374100e+06  5.247620e+05  8.596572e+05  1.264871e+06  

[122 rows x 33 columns]

* seeing 122 rows in the inner join - aka 122 samples that exist in both datasets
* seeing 135 rows in the left join - aka 135 samples that exist in the df dataset, fills unmatched HMO data with NaN

In [9]:
merged_2 = pd.merge(df, AC, on="sample_unique_id", how="left")
merged_2

Subject ID Patient Consented sample_unique_id   CGA  DOL   Current Weight  \
0      NB00012                 N      NB00012_M_1  28.4    27            1430   
1      NB00003                 Y      NB00003_M_1  29.0    22             995   
2      NB00003                 Y      NB00003_M_2  29.0    22            1005   
3      NB00003                 Y      NB00003_M_3  29.3    25            1065   
4      NB00354                 N      NB00354_M_1  29.4    12            1005   
..         ...               ...              ...   ...   ...             ...   
130    NB00007                 Y      NB00007_M_3  39.2    17            2340   
131    NB00007                 Y      NB00007_M_4  39.3    18            2380   
132    NB00379                 Y      NB00379_M_1  39.6    98            3073   
133    NB00007                 Y      NB00007_M_5  40.0    23            2430   
134    NB00007                 Y      NB00007_M_7  40.2    23            2400   

     Current Height  Current HC Scavenged/Fresh?   MBM/DMB?  ...  \
0              32.0        26.0       Scavenged   MBM + DBM  ...   
1              35.7        23.7       Scavenged         DBM  ...   
2              36.0        24.4       Scavenged         DBM  ...   
3              36.0        24.3       Scavenged         DBM  ...   
4              34.0        34.5       Scavenged         MBM  ...   
..              ...         ...              ...        ...  ...   
130            44.0        33.0       Scavenged         MBM  ...   
131            44.0        33.0       Scavenged         MBM  ...   
132            45.0        33.2       Scavenged         MBM  ...   
133            45.0        34.5       Scavenged        MBM   ...   
134            45.0        34.5       Scavenged        MBM   ...   

             LNnT         LNFPI        LNFPII      LNFPIII          LSTc  \
0    7.026127e+05  3.686662e+06  9.449716e+06  115974.6422  1.878166e+06   
1    3.554526e+05  2.188585e+06  2.289321e+06   75037.6026  4.920676e+05   
2    1.009692e+06  4.518359e+06  3.409862e+06   82939.5756  7.308408e+05   
3    6.291601e+05  4.624398e+06  3.044566e+06   95822.1100  6.416626e+05   
4    3.386165e+05  7.478814e+05  7.146520e+06  167202.2557  1.778718e+06   
..            ...           ...           ...          ...           ...   
130  6.844752e+05  6.025784e+06  4.126527e+06  174538.3580  1.167148e+06   
131  5.297594e+05  2.120913e+06  1.480291e+07  532044.6917  3.070643e+06   
132  6.555827e+05  1.314651e+07  1.863176e+06   84150.5867  1.410230e+06   
133  8.896175e+05  6.627449e+06  7.549175e+06  211982.9758  1.452051e+06   
134           NaN           NaN           NaN          NaN           NaN   

            DFLNT         DSLNT         DFLNH        FDSLNH         DSLNH  
0    2.210782e+06  1.358187e+06  1.686614e+06  2.734042e+06  3.015652e+06  
1    2.433375e+06  4.284901e+05  2.886016e+05  2.508040e+05  2.211292e+05  
2    2.628986e+06  5.605861e+05  6.504227e+05  5.128080e+05  5.097773e+05  
3    1.450763e+06  4.879326e+05  4.226243e+05  7.237869e+05  4.211949e+05  
4    9.524693e+05  1.128755e+06  1.396158e+06  2.356293e+06  2.505246e+06  
..            ...           ...           ...           ...           ...  
130  5.337920e+06  1.809488e+06  1.026485e+06  2.662468e+05  8.587586e+05  
131  1.352836e+06  1.923140e+06  5.688529e+05  1.727998e+06  1.916662e+06  
132  5.056181e+04  5.865819e+05  5.460105e+05  1.194098e+05  1.291120e+06  
133  6.723469e+06  2.374100e+06  5.247620e+05  8.596572e+05  1.264871e+06  
134           NaN           NaN           NaN           NaN           NaN  

[135 rows x 33 columns]

In [ ]:
#list of IDs in the metadata but not the AC file

set(df['sample_unique_id']) - set(AC['sample_unique_id'])


{'NB00001_M_8',
 'NB00002_M_7',
 'NB00003_M_24',
 'NB00003_M_25',
 'NB00003_M_5',
 'NB00007_M_7',
 'NB00010_M_4',
 'NB00013_M_1',
 'NB00015_M_1',
 'NB00380_M_3',
 'NB00398_M_7',
 'NB00423_M_1',
 'NB00457_2'}

In [ ]:
#list of IDs in the AC file but not in the metadata

set(AC['sample_unique_id']) - set(df['sample_unique_id'])


{'BN00380_M_3',
 'NB00001_M_2',
 'NB00002_M_9 ?',
 'NB00007_M_6',
 'NB00008_M_6',
 'NB00013_M1',
 'NB00015_M1',
 'NB00101_M_1',
 'NB0010_M_4',
 'NB00361_M_6',
 'NB00361_M_7',
 'NB00457_M_2',
 'NB0398_M_7'}